blueflood-core/src/main/java/com/rackspacecloud/blueflood/rollup/Granularity.java

```diff
@@ -274,9 +288,29 @@ private static Granularity granularityFromPointsGeometric(double requestedDurati
                 break;
             }
         }
+
+        if (gran == null) {
+            gran = Granularity.LAST;
+        }
+
         return gran;
     }
 
+    private static long getTTL(String tenantid, Granularity g) {
+        long ttl;
+        try {
+            ttl = (long)TTL_PROVIDER.getTTL(
```

Comment:
@chinmay-gupte points out that this is not correct for full res metrics.  See here:
https://github.com/rackerlabs/blueflood/blob/master/blueflood-http/src/main/java/com/rackspacecloud/blueflood/inputs/handlers/HttpMetricsIngestionHandler.java#L178